server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java

```diff
@@ -353,58 +313,159 @@ private void loadProps() {
-      if (TSFileDescriptor.getInstance().getConfig().getPageSizeInByte() > TSFileDescriptor
-          .getInstance().getConfig().getGroupSizeInByte()) {
-        logger
-            .warn("page_size is greater than group size, will set it as the same with group size");
-        TSFileDescriptor.getInstance().getConfig()
-            .setPageSizeInByte(TSFileDescriptor.getInstance().getConfig().getGroupSizeInByte());
-      }
-      TSFileDescriptor.getInstance().getConfig().setMaxNumberOfPointsInPage(Integer
-          .parseInt(properties.getProperty("max_number_of_points_in_page",
-              Integer.toString(
-                  TSFileDescriptor.getInstance().getConfig().getMaxNumberOfPointsInPage()))));
-      TSFileDescriptor.getInstance().getConfig().setTimeSeriesDataType(properties
-          .getProperty("time_series_data_type",
-              TSFileDescriptor.getInstance().getConfig().getTimeSeriesDataType()));
-      TSFileDescriptor.getInstance().getConfig().setMaxStringLength(Integer
-          .parseInt(properties.getProperty("max_string_length",
-              Integer.toString(TSFileDescriptor.getInstance().getConfig().getMaxStringLength()))));
-      TSFileDescriptor.getInstance().getConfig().setBloomFilterErrorRate(Double
-          .parseDouble(properties.getProperty("bloom_filter_error_rate",
-              Double.toString(
-                  TSFileDescriptor.getInstance().getConfig().getBloomFilterErrorRate()))));
-      TSFileDescriptor.getInstance().getConfig().setFloatPrecision(Integer
-          .parseInt(properties
-              .getProperty("float_precision", Integer
-                  .toString(TSFileDescriptor.getInstance().getConfig().getFloatPrecision()))));
-      TSFileDescriptor.getInstance().getConfig().setTimeEncoder(properties
-          .getProperty("time_encoder",
-              TSFileDescriptor.getInstance().getConfig().getTimeEncoder()));
-      TSFileDescriptor.getInstance().getConfig().setValueEncoder(properties
-          .getProperty("value_encoder",
-              TSFileDescriptor.getInstance().getConfig().getValueEncoder()));
-      TSFileDescriptor.getInstance().getConfig().setCompressor(properties
-          .getProperty("compressor", TSFileDescriptor.getInstance().getConfig().getCompressor()));
+      loadTsFileProps(properties);
 
+    } catch (FileNotFoundException e) {
+      logger.warn("Fail to find config file {}", url, e);
     } catch (IOException e) {
-      logger.warn("Cannot load config file because, use default configuration", e);
+      logger.warn("Cannot load config file, use default configuration", e);
     } catch (Exception e) {
       logger.warn("Incorrect format in config file, use default configuration", e);
     } finally {
       // update all data seriesPath
       conf.updatePath();
-      try {
-        inputStream.close();
-      } catch (IOException e) {
-        logger.error("Fail to close config file input stream because ", e);
```

Comment:
By putting the initialization of InputStream into try statement, will it automatically close the stream and catch the IOException during closing?

Reply:
Yes